lib/src/main/java/io/ably/lib/rest/AblyRest.java

```diff
@@ -149,6 +152,51 @@ public void statsAsync(Param[] params, Callback<AsyncPaginatedResult<Stats>> cal
 	}
 
 	/**
+	 * Make a generic HTTP request for a collection of the given type
+	 * @param params query options: see Ably REST API documentation
+	 * for available options
+	 * @return a PaginatedResult of Stats records for the requested params
+	 * @throws AblyException
+	 */
+	public PaginatedResult<JsonElement> paginatedRequest(String method, String path, Param[] params, RequestBody body, Param[] headers) throws AblyException {
```

Comment:
If we cannot support both a paginated array of results and a single JsonElement in a single method, I propose we instead name them according to the original method prefix i.e. `requestObject` and `requestPaginated` so that at least anyone expecting the `request` method would find them in intellisense.  I am not sure if `requestObject` is right, but I don't like the idea that `request` implements the spec somewhat and relies on the user discovering `paginatedRequest`. Instead I'd prefer that the user finds neither and discovers why there are two variations instead.